pkg/config/config_test.go

```diff
@@ -189,47 +187,21 @@ func TestStorageEnvOverrides(t *testing.T) {
 		},
 	}
 	envVars := getEnvMap(&Config{Storage: expStorage})
-	envVarBackup := map[string]string{}
 	for k, v := range envVars {
-		oldVal := os.Getenv(k)
-		envVarBackup[k] = oldVal
-		os.Setenv(k, v)
+		t.Setenv(k, v)
 	}
 	conf := &Config{}
 	err := envOverride(conf)
 	if err != nil {
 		t.Fatalf("Env override failed: %v", err)
 	}
 	compareStorageConfigs(conf.Storage, expStorage, t)
-	restoreEnv(envVarBackup)
 }
 
 // TestParseExampleConfig validates that all the properties in the example configuration file
 // can be parsed and validated without any environment variables
 func TestParseExampleConfig(t *testing.T) {
-
-	// initialize all struct pointers so we get all applicable env variables
-	emptyConf := &Config{
-		Storage: &Storage{
-			Disk: &DiskConfig{},
-			GCP:  &GCPConfig{},
-			Minio: &MinioConfig{
-				EnableSSL: false,
-			},
-			Mongo: &MongoConfig{},
-			S3:    &S3Config{},
-		},
-		SingleFlight: &SingleFlight{},
-		Index:        &Index{},
-	}
-	// unset all environment variables
-	envVars := getEnvMap(emptyConf)
-	envVarBackup := map[string]string{}
-	for k := range envVars {
-		oldVal := os.Getenv(k)
-		envVarBackup[k] = oldVal
-		os.Unsetenv(k)
-	}
+	os.Clearenv()
```

Comment:
CMIIW but this won't restore the env after the test is done, right?

Would something like this work?

```
for k := range envVars {
    t.Setenv(k, "")
    os.Unsetenv(k)
}
```

Reply:
Yes, you are right.

For `TestParseExampleConfig`, it is not necessary to restore the environment variables. The aim of this PR is to ensure that none of the tests in `config_test.go` depend on one or more environment variables set by another test.

Actually, `os.Clearenv()` is also optional. None of our tests have leftover environment variables so I wrote `os.Clearenv()` here as a fail-safe.

Reply:
Right, makes sense. Thanks!